server/base/src/main/java/org/apache/accumulo/server/conf/CheckCompactionConfig.java

```diff
@@ -0,0 +1,180 @@
+import org.apache.accumulo.core.spi.compaction.ExecutorManager;
+import org.apache.accumulo.core.util.compaction.CompactionExecutorIdImpl;
+import org.apache.accumulo.start.spi.KeywordExecutable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.beust.jcommander.Parameter;
+import com.google.auto.service.AutoService;
+import com.google.common.collect.Iterables;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+@AutoService(KeywordExecutable.class)
+public class CheckCompactionConfig implements KeywordExecutable {
+
+  private final static Logger log = LoggerFactory.getLogger(CheckCompactionConfig.class);
+
+  static class Opts extends Help {
+    @Parameter(description = "<path/to/props/file>")
+    String filePath;
+  }
+
+  @Override
+  public String keyword() {
+    return "check-compaction-config";
+  }
+
+  @Override
+  public String description() {
+    return "Checks compaction config";
+  }
+
+  public static void main(String[] args) throws IOException {
+    new CheckCompactionConfig().execute(args);
+  }
+
+  @SuppressFBWarnings(value = "DM_EXIT",
+      justification = "System.exit is fine here because it's a utility class executed by a main()")
+  @Override
+  public void execute(String[] args) throws IOException {
+    Opts opts = new Opts();
+    opts.parseArgs("accumulo check-compaction-config", args);
+
+    if (opts.filePath == null) {
+      System.err.println("No properties file was given");
+      System.exit(1);
+    }
+
+    Path path = Path.of(opts.filePath);
+    if (!path.toFile().exists()) {
+      System.err.println("File at given path could not be found");
+      System.exit(1);
+    }
+
+    // Extract properties from props file at given path
+    Properties allProps = ClientInfoImpl.toProperties(path);
+    log.debug("All props: {}", allProps);
+
+    // Extract server props from set of all props
+    Map<String,String> serverPropsMap = new HashMap<>();
+    String suffix = ".accumulo.server.props";
+    allProps.forEach((k, v) -> {
+      if (k.toString().endsWith(suffix))
+        serverPropsMap.put(k.toString(), v.toString());
+    });
+
+    // Ensure there is exactly one server prop in the map and get its value
+    // The value should be compaction properties
+    String compactionPropertiesString = "";
+    try {
+      compactionPropertiesString = Iterables.getOnlyElement(serverPropsMap.values());
+    } catch (IllegalArgumentException e) {
+      System.err.println("There should be exactly one server property. " + e.getMessage());
+      System.exit(1);
+    }
+
+    // Create a props object for compaction props
+    StringReader sr = new StringReader(compactionPropertiesString.replace(' ', '\n'));
+    Properties serverProps = new Properties();
+    serverProps.load(sr);
+
+    // Extract executors options from compactions props
+    Map<String,String> executorsProperties =
+        getPropertiesWithSuffix(serverProps, ".planner.opts.executors");
```

Comment:
Multiple compaction services can be configured, each with its own planner and options.

Reply:
Whoops, good catch, ill make sure to incorporate that.

Reply:
Addressed in 4232ea1